### ISSUE TITLE
Typed API: Get and document not found

### ISSUE THREAD

**Issue by rbcervilla** [elastic/go-elasticsearch#678]
When I try to get a document that doesn't exist, returned values dont't make sense.

**Example**
```go
res, err := es.Get("index_name", "id_not_exists").Do(context.Background())
```
Returned values are a nil response and error ElasticsearchError with Status 0

I think returned values should be nil error with Response
```go
Response{
    Index_: "index_name",
    Id_: "id_not_exists",
    Found: false,
}
```
Or maybe a nil response and ElasticsearchError with Status 404

Is it correct?

**Comment by Anaethelion** [comment]
Thank you for reporting this.

Your reasoning is correct and it seems the [specification](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_global/get/GetResponse.ts) is missing this exceptions in the `Get` response.

I'll add that and update this issue when the API has been regenerated!

**Comment by Anaethelion** [comment]
Fixed in release [release/8.8.2](https://github.com/elastic/go-elasticsearch/releases/tag/v8.8.2)
